test/model.create.test.js

```diff
@@ -197,6 +197,27 @@ describe('model', function() {
         const docs = await Test.find();
         assert.equal(docs.length, 5);
       });
+      it('should throw an error only after all the documents have finished saving gh-4628', async function() {
+        const countSchema = new Schema({ n: Number });
+        const testSchema = new Schema({ name: { type: String, unique: true }, reference: Number });
+
+        const Count = db.model('gh4628', countSchema);
```

Comment:
Missing `await Count.init()` after this, see https://mongoosejs.com/docs/validation.html#the-unique-option-is-not-a-validator.